simulator/infra/t/unit_test.cpp

```diff
@@ -275,3 +275,19 @@ TEST_CASE("Test uint128 octal printing")
     CHECK( uint128_to_oct_string( narrow_cast<uint128>(UINT64_MAX) + 1) == "2000000000000000000000");
     CHECK( uint128_to_oct_string( all_ones<uint128>()) == "3777777777777777777777777777777777777777777");
 }
+
+TEST_CASE("Test uint32 circular left shift")
+{
+    uint32 value = 0xA0B0'C0D0;
+    CHECK( value == circ_ls( value, 0));
+    CHECK( value == circ_ls( value, 32));
```

Comment:
BTW, our `macro.h` tests usually assume that `shamt` is "normalized" already, so you probably don't need to test it. But for zero, special handling is required.